exp-player/tests/unit/randomizers/random-parameter-set-test.js

```diff
@@ -0,0 +1,161 @@
+import Ember from 'ember';
+
+import { module } from 'qunit';
+import test from 'dummy/tests/ember-sinon-qunit/test';
+
+import { getRandomElement, replaceValues, randomizer } from 'exp-player/randomizers/random-parameter-set';
+
+module('Unit | Randomizer | random parameter set');
+
+test('Random element selected from weighted list is a possible choice', function (assert) {
+    const arr       = [1, 2, 3, 4, 5, 6, 7, 8];
+    const weights   = [0, 0, 0, 0, 0, 1, 0, 0];
+
+    const expectedResult = 6;
+
+    let [index, actualResult] = getRandomElement(arr, weights);
+
+    assert.deepEqual(actualResult, expectedResult,
+        'Random element selected should not have probability weight 0'
+    );
+});
+
+test('Values are replaced at multiple levels of object hierarchy and within arrays', function (assert) {
+
+    const object = {
+        prop0: 'val4',
+        prop1: {
+            setting1: 1,
+            setting2: "val1",
+            setting3: "val1",
+            setting4: [4,"val3",6],
+            setting5: 5
+        },
+        prop2: {
+            setting1: {
+                subSetting: "val2"
+            },
+            setting2: "val3"
+        },
+        prop3: {
+            setting: "val4"
+        }
+    };
+
+    const replace = {
+        "val1": "replacedvalue1",
+        "val2": "replacedvalue2",
+        "val3": "replacedvalue3",
+        "val4": "replacedvalue4"
+    };
+
+    const expectedResult = {
+        prop0: 'replacedvalue4',
+        prop1: {
+            setting1: 1,
+            setting2: "replacedvalue1",
+            setting3: "replacedvalue1",
+            setting4: [4,"replacedvalue3",6],
+            setting5: 5
+        },
+        prop2: {
+            setting1: {
+                subSetting: "replacedvalue2"
+            },
+            setting2: "replacedvalue3"
+        },
+        prop3: {
+            setting: "replacedvalue4"
+        }
+    };
+
+    let actualResult = replaceValues(object, replace);
+
+    assert.deepEqual(actualResult, expectedResult,
+        'Strings that are properties of replace should be replaced throughout object'
+    );
+});
+
+
+// Object.assign doesn't work in tests; omitting this for now (but have checked manually)
```

Comment:
This probably has to do with how we transpile code.

Excited to see some tests around the randomizer; thanks!

Per this note, I believe that `Ember.assign` might provide a useful wrapper/ polyfill for your needs here...? Basically, the tool that lets us pretend to write modern javascript (es6+) doesn't automatically add *all* of the new stuff in by default, at least when we're running tests. So ember offers some selected helpers.*

Incidentally: to help track tests that need attention later, you can replace `test()` with `skip()`- that's generally a handy alternative to commenting out code.


** The fancy-schmancy terms are Babel, transpile, and polyfill. There are [ways](https://github.com/CenterForOpenScience/ember-preprints/blob/develop/ember-cli-build.js#L103) to control the polyfills used, but we try not to have to use them unless absolutely necessary.